gap/generic/SnAnUnknownDegree.gi

```diff
@@ -0,0 +1,895 @@
+#############################################################################
+##
+##  This file is part of recog, a package for the GAP computer algebra system
+##  which provides a collection of methods for the constructive recognition
+##  of groups.
+##
+##  This files's authors include Friedrich Rober and Sergio Siccha.
+##
+##  Copyright of recog belongs to its developers whose names are too numerous
+##  to list here. Please refer to the COPYRIGHT file for details.
+##
+##  SPDX-License-Identifier: GPL-3.0-or-later
+##
+##
+##  This file provides code for recognising whether a permutation group
+##  is isomorphic to an alternating or symmetric group. It implements
+##  <Cite Key="JLNP13"/>.
+##
+#############################################################################
+#
+# FIXME: Move me into GAP
+# Helper function to compute all primes up to a given integer via a prime sieve
+RECOG.AllPrimesUpTo := function(n)
+    local i, j, sieve, result;
+    if n <= 1000 then
+      return Primes{[1..PositionSorted(Primes, n+1)-1]};
+    fi;
+    sieve := BlistList([1..n], [1..n]);
+    sieve[1] := false;
+    for i in [2..Int(n/2)] do
+        sieve[i*2] := false;
+    od;
+    i := 3;
+    while i * i <= n do
+        if sieve[i] then
+            j := 3*i;
+            while j <= n do
+                sieve[j] := false;
+                j := j + 2*i;
+            od;
+        fi;
+        i := i + 2;
+    od;
+    return ListBlist([1..n], sieve);
+end;
+
+# eps : real number, the error bound
+# N : integer, upper bound for the degree of G
+#
+# Returns a record of constants used in ThreeCyclesCanditatesIterator.
+RECOG.ThreeCycleCandidatesConstants := function(eps, N)
+    local M, allPrimes, i, p;
+    # Constants
+    M := 1;
+    allPrimes := RECOG.AllPrimesUpTo(N);
+    for i in [2 .. Length(allPrimes)] do
+        p := allPrimes[i];
+        M := M * p ^ LogInt(N, p);
+    od;
```

Comment:
With my suggestions above, this would become:
```suggestion
    RECOG.CachePrimesUpTo(N);
    for p in RECOG.PrimesCache do
        if p > N then break; fi;
        M := M * p ^ LogInt(N, p);
    od;
```